datafusion/common/src/utils.rs

```diff
@@ -390,6 +390,16 @@ pub fn arrays_into_list_array(
     ))
 }
 
+/// Get the child arrays from a `ArrayRef`.
+pub fn array_into_children_array_vec(list_arr: &ArrayRef) -> Vec<ArrayRef> {
```

Comment:
I think this handles nulls and sliced arrays incorrectly. Perhaps you could just use the comparison kernels on the lists directly, instead of partially decomposing them?

Reply:
```text
Err(InvalidArgumentError("Invalid comparison operation: List(Field { name: \"item\", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }) < List(Field { name: \"item\", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} })"))
```

```rust
                    let lt_res =
                            arrow::compute::kernels::cmp::lt(&arr1, &arr2);
```

Reply:
In which case do we need to support this on scalar value at all?

Reply:
> In which case do we need to support this on scalar value at all?

Probably none. I support it since it has supported before.

Reply:
Let me split it to two function for now.